### PR TITLE
SW-1269 usage file and backup file will break if version number is not text only

### PR DESCRIPTION
**Pull request by JFSolution**
- repair broken backup file
- convert unicode version number to str

### CODE REVIEW

File: octoprint_mrbeam/analytics/usage_handler.py [hunk 1]
```diff
@@ -356,36 +358,59 @@ def _load_usage_data(self):
...
-            if os.path.isfile(self._backup_file):
+            try:
+                with open(self._backup_file, "r") as stream:
+                    data = yaml.safe_load(stream)
+                if self._validate_data(data):
+                    data["restored"] = (
+                        data["restored"] + 1 if "restored" in data else 1
+                    )
+                    self._usage_data = data
+                    success = True
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/analytics/usage_handler.py]: is it safe to set this variable to True before writing to the file ?

File: octoprint_mrbeam/analytics/usage_handler.py [hunk 2]
```diff
@@ -356,36 +358,59 @@ def _load_usage_data(self):
...
+            except:
+                self._logger.error("Can't read _backup_file file.")
 
         if not success:
             self._logger.warn("Resetting usage data. (marking as incomplete)")
             self._usage_data = self._get_usage_data_template()
             if recovery_try:
                 self._write_usage_data()
 
+    def _repair_backup_usage_data(self):
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/analytics/usage_handler.py:390]: Please add method doc string

File: octoprint_mrbeam/analytics/usage_handler.py [hunk 3]
```diff
@@ -356,36 +358,59 @@ def _load_usage_data(self):
...
+        success = False
+        with open(self._backup_file, "r") as stream:
+            data = yaml.load(stream)
+        if self._validate_data(data):
+            if isinstance(data["version"], unicode):
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/analytics/usage_handler.py:402]: Please add a comment explaining why we need this and put the link to the Jira issue in the comment as well

File: octoprint_mrbeam/analytics/usage_handler.py [hunk 4]
```diff
@@ -356,36 +358,59 @@ def _load_usage_data(self):
...
+                data["version"] = unicodedata.normalize('NFKD', data["version"]).encode('ascii', 'ignore')
+            data["restored"] = (
+                data["restored"] + 1 if "restored" in data else 1
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/analytics/usage_handler.py:405]: what is that line supposed to do ?

File: octoprint_mrbeam/analytics/usage_handler.py (hunk 1 above)
Comment (JFSolution) [reply octoprint_mrbeam/analytics/usage_handler.py]: I can change the order

File: octoprint_mrbeam/analytics/usage_handler.py (hunk 2 above)
Comment (JFSolution) [reply octoprint_mrbeam/analytics/usage_handler.py:390]: ok, will do

File: octoprint_mrbeam/analytics/usage_handler.py (hunk 3 above)
Comment (JFSolution) [reply octoprint_mrbeam/analytics/usage_handler.py:402]: ok

File: octoprint_mrbeam/analytics/usage_handler.py (hunk 4 above)
Comment (JFSolution) [reply octoprint_mrbeam/analytics/usage_handler.py:405]: the restored? It will count how often this file was restored from the backup file

**Review by Yelko-mrbeam — APPROVED** [review]

**Pull request merged**